### PR TITLE
Upgrade connect-mongo to 0.7.0 to support Mongoose 4.0

### PR DESCRIPTION
**Pull request by darko-mijic**
Upgraded connect-mongo to 0.7.0, old version (0.4.1) does work with Mongoose 4.0.0-rc3. 

API for reusing Mongoose connection has changed, for that resason config/lib/express.js and config/lib/socket.io.js are updated.

### CODE REVIEW

File: config/env/development.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 'use strict';
 
 module.exports = {
-	db: 'mongodb://localhost/mean-dev',
+    db: {
+        url: 'mongodb://localhost/mean-dev',
+        debug: true
```
Comment (ilanbiala) [inline config/env/development.js]: Why debug true..?

File: config/lib/express.js [hunk 2]
```diff
@@ -112,10 +112,10 @@ module.exports.initSession = function (app, db) {
 		saveUninitialized: true,
 		resave: true,
 		secret: config.sessionSecret,
-		store: new MongoStore({
-			db: db.connection.db,
-			collection: config.sessionCollection
-		})
+        store: new MongoStore({
```
Comment (ilanbiala) [inline config/lib/express.js]: Indentation.

Comment (darko-mijic) [inline config/lib/express.js]: @ilanbiala I can see that in come files tabs are used for indentation and in come files spaces. What is currently preferred?

Comment (ilanbiala) [inline config/lib/express.js]: Tabs please. Abide by the editorconfig and jshintrc.

File: config/env/development.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 module.exports = {
-	db: 'mongodb://localhost/mean-dev',
+	db: {
+		url: 'mongodb://localhost/mean-dev'
```
Comment (ilanbiala) [inline config/env/development.js]: please change it to `db: 'mongodb://localhost/mean-dev` to be consistent with 0.4.0.

**Comment by lirantal** [comment]
@ilanbiala I guess we can merge this in if LGTY.

**Comment by ilanbiala** [comment]
@darko-mijic can you just squash the commits so we don't have unnecessary bloat?

**Comment by darko-mijic** [comment]
@ilanbiala is this OK?

**Comment by ilanbiala** [comment]
1 commit with a simple message like "Upgrade connect-mongo to 0.7.0" should be sufficient.

**Comment by ilanbiala** [comment]
@darko-mijic If you can do `git rebase -i HEAD~5` and change the bottom 4 to `squash` instead of `pick`. Then we should be able to merge this in.

**Comment by ilanbiala** [comment]
Closing this because a PR against 0.4.0 has been created #498.

**Pull request closed without merging**